contrib/babelfishpg_tds/src/backend/tds/tdsprotocol.c

```diff
@@ -711,3 +725,31 @@ TestGetTdsRequest(uint8_t reqType, const char *expectedStr)
 	}
 	return res;
 }
+
+/*
+ * Start statement timeout timer, if enabled.
+ */
+static void
+enable_statement_timeout(void)
+{
+	if (StatementTimeout > 0)
+	{
+		if (!get_timeout_active(STATEMENT_TIMEOUT))
```

Comment:
what does this mean? if timer is already running do not do anything, shouldn't we reset the timeout to guc value

Reply:
Why is it needed to reset the guc value ?